src/core/load_balancing/xds/xds_cluster_impl.cc

```diff
@@ -209,6 +209,10 @@ class XdsClusterImplLb final : public LoadBalancingPolicy {
           [](RefCountedStringValue locality) { return locality; },
           [](const RefCountedPtr<LrsClient::ClusterLocalityStats>&
                  locality_stats) {
+            if (locality_stats == nullptr ||
```

Comment:
I don't think this should ever happen, so this doesn't seem like the right fix.

Instead, I think we need to understand why locality_stats is being passed down as null in the first place, since that should never be happening.

Reply:
I see, I missed that conversation on https://github.com/grpc/grpc/pull/36841. Let me tag you on the bug, you seem to be the right assignee here